src/main/java/budgetbuddy/logic/rules/RuleProcessingUtil.java

```diff
@@ -27,9 +67,12 @@ public static Object extractAttribute(Attribute attribute, Transaction txn) {
         switch (attribute) {
         case DESCRIPTION:
             return txn.getDescription();
-        case AMOUNT:
-            long sign = txn.getDirection().equals(Direction.IN) ? 1 : -1;
-            return sign * txn.getAmount().toLong();
+        case IN_AMOUNT:
+            return (txn.getDirection().equals(Direction.IN) ? 1 : -1)
+                    * (txn.getAmount().toLong() / 100);
+        case OUT_AMOUNT:
+            return (txn.getDirection().equals(Direction.OUT) ? 1 : -1)
+                    * (txn.getAmount().toLong() / 100);
```

Comment:
What's up with the `/ 100`? If you want to display the amount in dollars and cents, I think amount's `toString` does that already.

Reply:
I need the long value of it that's why. It's not for display but for calculation

Reply:
Ah alright. You might want to leave it in cents though, `long` doesn't support decimals.

Reply:
oh crap. Thanks for letting me know. I'll convert to double then